test/modules/packages/distributedIters/DistributedIters.chpl

```diff
@@ -0,0 +1,398 @@
+  if isDomain(c) then
+  {
+    assert(c.rank > 0, ("DistributedIters: Guided iterator (leader): Must "
+                        + "use a valid domain"));
+    assert(parDim > 0, ("DistributedIters: Guided iterator (leader): parDim "
+                        + "must be a positive integer"));
+    assert(parDim <= c.rank, ("DistributedIters: Guided iterator (leader): "
+                              + "parDim must be a dimension of the domain"));
+
+    var parDimDim = c.dim(parDim);
+
+    for t in distributedGuided(tag=iterKind.leader,
+                               c=parDimDim,
+                               numTasks=numTasks,
+                               parDim=1,
+                               minChunkSize=minChunkSize,
+                               coordinated=coordinated,
+                               workerLocales=workerLocales)
+    {
+      // Set the new range based on the tuple the guided 1-D iterator yields.
+      var newRange = t(1);
+
+      type cType = c.type;
+      // Does the same thing as densify, but densify makes a stridable domain,
+      // which mismatches here if c (and thus cType) is non-stridable.
+      var tempDom : cType = computeZeroBasedDomain(c);
+
+      // Rank-change slice the domain along parDim
+      var tempTup = tempDom.dims();
+      // Change the value of the parDim elem of the tuple to the new range
+      tempTup(parDim) = newRange;
+
+      yield tempTup;
+    }
+  }
+  else // c is a range.
+  {
+    const iterCount = c.length;
+
+    if iterCount == 0 then halt("DistributedIters: Guided iterator (leader): ",
+                                "the range is empty");
+
+    type cType = c.type;
+    const denseRange:cType = densify(c,c);
+
+    if iterCount == 1
+       || numTasks == 1 && numLocales == 1
+    then
+    {
+      if debugDistributedIters
+      then writeln("DistributedIters: Guided iterator (leader): serial ",
+                   "execution due to insufficient work or compute resources");
+      yield (denseRange,);
+    }
+    else
+    {
+      const numWorkerLocales = workerLocales.size;
+      const denseRangeHigh:int = denseRange.high;
+      const masterLocale = here.locale;
+
+      const potentialWorkerLocales =
+        [L in workerLocales] if numLocales == 1
+                                || !coordinated
+                                || L != masterLocale
+                             then L;
+      /*
+        It's not sensible to use a single locale besides masterLocale, so use
+        potentialWorkerLocales only if it's larger than one locale.
+      */
+      const actualWorkerLocales = if potentialWorkerLocales.size > 1
+                                  then potentialWorkerLocales
+                                  else [masterLocale];
+      const numActualWorkerLocales = actualWorkerLocales.size;
+
+      // The guided iterator stage (determines next subrange index and size).
+      var meitneriumIndex:atomic int;
```

Comment:
You're probably right... I'll argue unsuccessfully that I got the idea from learnChapelInYMinutes's atomic int named uranium. :)